src/components/molecules/MapLegend.vue

```diff
@@ -1,259 +1,363 @@
 <template>
-  <div 
-    v-if="hasActiveLayer" 
-    class="Legend mb-3">
-    <div
-      class="Legend__title d-flex pl-3 ml-3">
-      <span class="align-self-center">
-        LEGENDA
-      </span>
+  <div v-if="hasActiveBundle" class="Legend mb-3">
+    <div class="Legend__title d-flex pl-3 ml-3">
+      <span class="align-self-center"> LEGENDA </span>
     </div>
-    <div class="Legend__subtitle d-flex pl-3 ml-3 mt-2">
-      <span class="align-self-center">
-        {{ legendLabel }}
-      </span>
-    </div>
-    <div>
-      <ul class="m-0 p-0 pl-3 ml-3 mb-3 list-unstyled">
-        <li 
-          v-for="(item, index) in legendItems" 
-          :key="index"
-          class="d-flex my-2 mr-3 align-items-center">
-          <span class="Legend__point" :style="{ backgroundColor: item.color }"></span> 
-          <span class="mx-2 flex-grow-1">{{ item.label }}</span>
-        </li>
-      </ul>
+    <div v-for="layer in activeLayers" :key="layer.id">
+      <div class="Legend__subtitle d-flex pl-3 ml-3 mt-2">
+        <span class="px-2" @click="toggleVisibility(layer)">
+          <img
+            :src="
+              getLayerVisibility(layer)
+                ? icon('Eye-regular-icon.svg')
+                : icon('Eye-slash-regular-icon.svg')
+            "
+            width="14"
+            height="14"
+          />
+        </span>
+        <span class="align-self-center">
+          {{ layer.name }}
+        </span>
+      </div>
+      <div v-if="getLayerVisibility(layer)">
+        <ul class="m-0 p-0 pl-3 ml-3 mb-3 list-unstyled">
+          <li
+            v-for="(item, index) in generateLegendForLayer(layer)"
```

Comment:
Computed values don't take arguments (they can, but its a hack). Since we request the visibility for a specific layer in our collection we need to pass this arg and therefore use a method.